### DISCUSSION TITLE
Purpose of 'WSCRIPT.exe' Component

### DISCUSSION THREAD

**Discussion by JDot747** [Romanitho/Winget-AutoUpdate#478] [General]
Hello,

Awesome app, thanks for maintaing it.

I was wondering what the functionality of 'wscript.exe' was within WAU?

Would WAU be successful if wscript was blocked?

The reason I ask is that we are currently blocking wscript.exe as per Microsoft's recommendation. Trying to determine if WAU would still be effective with this blocked.
- https://learn.microsoft.com/en-us/windows/security/application-security/application-control/windows-defender-application-control/design/applications-that-can-bypass-wdac

**Comment by JDot747** [comment]
Completed some initial testing.

Am finding that the 'User Context' updates don't seem to work. The system context updates will run and then it attempts the user context as this is shown in the log '14:50:16 - Starting WAU in User context'. Nothing happens after that.

When running the Scheduled Task manually it simply fails. The history shows 'wscript.exe' instances failed top start.

**Comment by KnifMelti** [comment]
Wscript is only used to run ps totally silent #465 
I'm curious, do you block all of the applications on that list?

**Reply by KnifMelti** [reply]
...and for creating shortcuts if that is enabled when installing

**Reply by Romanitho** [reply]
to be a bit more technical, PowerShell cannot be totally hidden if run interactively. You can test it by running PS script with `-WindowStyle hidden` you'll see a PowerShell window flash. To workaround this, we use a VBS caller. That's why `wscript.exe` is used.
basically, you can use WAU without it, user experience will not be as clean as it currently is.

as mentioned by @KnifMelti, #465 we are looking for another solution to accomplish the same process without vbs.

**Reply by JDot747** [reply]
Thanks for the info.

@KnifMelti; Yes, we block all applications on that list.
